Fixed maddening "..." lately in printed types.

paulp opened the pull request:
This must have been me when I eliminated some of the
remaining distinction between TypeRef(_, moduleClass, Nil) and
SingleType(_, moduleClass).  Sorry I didn't track it down sooner.
Review by anyone who is around because this is overdue.

paulp committed:
Fixed maddening "..." lately in printed types.

This must have been me when I eliminated some of the
remaining distinction between TypeRef(_, moduleClass, Nil) and
SingleType(_, moduleClass).  Sorry I didn't track it down sooner. Review by anyone who is around because this is overdue.

scala-jenkins commented:
Started jenkins job pr-rangepos at https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/20/

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/718/

scala-jenkins commented:
jenkins job pr-rangepos: Failed - https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/20/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Failed - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/718/

jsuereth commented:
```
/localhome/jenkins/b/workspace/pr-scala-testsuite-linux-opt/test/files/run/t6028.scala [FAILED]
```

Spurious?

paulp commented:
No, legit this time.  Working on it.

paulp committed:
Checkfile update.

I screwed around with type printing for a long time and now I have
to be done, so I suggest we accept the imperfection in here for now
because it's still way ahead of "..." as types go.

scala-jenkins commented:
Started jenkins job pr-rangepos at https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/24/

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/722/

hubertp commented:
I think it would be nice to refactor things like maxTostringRecursions to some single internal config file (similar to Settings) but that's not really the subject of this bug. lgtm.

scala-jenkins commented:
jenkins job pr-rangepos: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/24/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/722/
